packages/design-system/src/components/VerticalNav/VerticalNavItem.tsx

```diff
@@ -1,137 +1,195 @@
+    if (hasSubnav()) {
+      return handleToggleClick();
     }
 
-    return false;
-  }
+    return handleClick(evt);
+  };
 
   /**
    * Checks if a descendant is selected
    * @param {Array} children - The nested items
    * @return {Boolean}
    */
-  childIsSelected(children) {
+  const childIsSelected = (children): boolean => {
     if (children && children.length) {
-      return children.some((child) => {
-        return child.id === this.props._selectedId || this.childIsSelected(child.items);
+      return children.some((child: any) => {
+        return child.id === props._selectedId || childIsSelected(child.items);
       });
     }
 
     return false;
-  }
+  };
+
+  /**
+   * Check if this item is selected or if it is a parent of a selected item
+   * @return {Boolean}
+   */
+  const isSelected = (): boolean => {
+    if (props.selected) return props.selected;
+
+    if (props._selectedId && hasSubnav()) {
+      return childIsSelected(props.items);
+    }
+
+    return false;
+  };
 
-  subnavItems() {
-    if (this.props.url) {
+  const subnavItems = (): any => {
+    if (props.url) {
       // Since the VerticalNavItemLabel will just toggle the subnav, we
       // add a link to the top of the subnav for this item. Otherwise there
       // wouldn't be a way to actually visit its URL
-      const item = Object.assign({}, this.props);
+      const item = Object.assign({}, props);
       delete item.items;
 
-      return [item].concat(this.props.items);
+      return [item].concat(props.items);
     }
 
-    return this.props.items;
-  }
+    return props.items;
+  };
 
-  renderSubnav() {
-    if (this.hasSubnav()) {
+  const renderSubnav = (): React.ReactElement => {
+    if (hasSubnav()) {
       return (
         <VerticalNav
-          selectedId={this.props._selectedId}
-          collapsed={this.state.collapsed}
-          id={this.subnavId}
-          items={this.subnavItems()}
-          component={this.props.component}
+          selectedId={props._selectedId}
+          collapsed={collapsed}
+          id={subnavId}
+          items={subnavItems()}
+          component={props.component}
           nested
         />
       );
     }
-  }
-
-  render() {
-    const classes = classNames('ds-c-vertical-nav__item', this.props.className);
-
-    return (
-      <li className={classes}>
-        <VerticalNavItemLabel
-          ariaCollapsedStateButtonLabel={this.props.ariaCollapsedStateButtonLabel}
-          ariaExpandedStateButtonLabel={this.props.ariaExpandedStateButtonLabel}
-          collapsed={this.state.collapsed}
-          component={this.props.component}
-          label={this.props.label}
-          hasSubnav={this.hasSubnav()}
-          onClick={this.handleLabelClick}
-          selected={this.isSelected()}
-          subnavId={this.subnavId}
-          url={this.props.url}
-        />
-        {this.renderSubnav()}
-      </li>
-    );
-  }
-}
+  };
```

Comment:
This could also just become a value or be inlined at the bottom, since it is only called once. In the class components, it made sense to organize these into different functions, but now that it's all one function anyway, organizing them into functions isn't really necessary.